pkg/mover/intermediate.go

```diff
@@ -119,9 +129,113 @@ func tarImages(imageChanges []*internal.ImageChange, logger Logger) (string, err
 		logger.Printf("Processing image %s\n", change.ImageReference.Name())
 	}
 
-	logger.Printf("Writing all %d images...\n", len(refToImage))
+	logger.Printf("Writing %d images...\n", len(refToImage))
 	if err := tarball.MultiRefWrite(refToImage, imagesFile); err != nil {
 		return "", err
 	}
 	return imagesFile.Name(), nil
 }
+
+// IsIntermediateBundle returns tue only if VerifyIntermediateBundle finds no errors
+func IsIntermediateBundle(bundlePath string) bool {
+	return VerifyIntermediateBundle(bundlePath) == nil
+}
+
+type fileValidations struct {
+	filename, format string
+	validate         func(io.Reader) error
+}
+
+// VerifyIntermediateBundle returns true if the path points to an uncompressed
+// tarball with:
+//  A hints.yaml YAML file
+//  A manifest.json for the images
+//  A directory container an unpacked chart directory with valid YAMLs Chart.yaml & values.yaml
+func VerifyIntermediateBundle(bundlePath string) error {
+	validations := []fileValidations{
+		{filename: "hints.yaml", format: "YAML", validate: validateYAML},
+		{filename: originalChart + "/Chart.yaml", format: "YAML", validate: validateYAML},
+		{filename: originalChart + "/values.yaml", format: "YAML", validate: validateYAML},
+		{filename: imagesTar, format: "TAR", validate: validateTAR},
+	}
+	for _, fv := range validations {
+		r, err := openFromTar(bundlePath, fv.filename)
+		if err != nil {
+			return fmt.Errorf("failed to open file %s from tar: %w", fv.filename, err)
+		}
+		defer r.Close()
+		if err := fv.validate(r); err != nil {
+			return fmt.Errorf("%w: %s is not valid %s: %v",
+				ErrNotIntermediateBundle, fv.filename, fv.format, err)
+		}
+	}
+	return nil
+}
+
+type intermediateBundle struct {
+	bundlePath string
+}
+
+func openBundle(bundlePath string) (*intermediateBundle, error) {
```

Comment:
This method seems an unnecessary wrapper and in fact it does not open anything.

I'd just verify the bundle where you now call openBundle (a single place actually)

```
cm.bundle, err = openBundle(bundlePath)
	if err != nil {
		return err
	}
```

Reply:
ok, makes sense. Open is just validating.

Reply:
Actually we need to validate on the cmd, because that decides how we build the request.

here I can skip altogether.

Reply:
which is kind of unfair for direct API usage, because then the validation is never run before consumption on that case.

Reply:
openBundle removed now I have a simple newBundle